feature-libs/my-account/organization/components/unit/services/unit-list.service.ts

```diff
@@ -1,76 +1,136 @@
 import { Injectable } from '@angular/core';
-import { B2BUnit, B2BUnitNode, EntitiesModel } from '@spartacus/core';
+import { B2BUnitNode, B2bUnitTreeNode, EntitiesModel } from '@spartacus/core';
 import { OrgUnitService } from '@spartacus/my-account/organization/core';
 import { TableService } from '@spartacus/storefront';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { first, map, switchMap, withLatestFrom } from 'rxjs/operators';
 import { OrganizationListService } from '../../shared/organization-list/organization-list.service';
 import { OrganizationTableType } from '../../shared/organization.model';
+import { UnitItemService } from './unit-item.service';
 
 /**
- * Service to populate Budget data to `Table` data. Budget
- * data is driven by the table configuration, using the `OrganizationTables.BUDGET`.
+ * Service to populate Unit data to `Table` data. Unit
+ * data is driven by the table configuration, using the `OrganizationTables.UNIT`.
  */
 @Injectable({
   providedIn: 'root',
 })
-export class UnitListService extends OrganizationListService<B2BUnit> {
+export class UnitListService extends OrganizationListService<B2bUnitTreeNode> {
```

Comment:
While reviewing I've spotted a few issues/limitations with the unit tree:
- (issue) The active tree item is not expanded if you navigate browser history
- (issue) The active item is also expanded, but it shouldn't imho. we should only open the tree until the active item instead. 
- (limitation) We could think of collapsing/expanding _all_ items for a specific part of the tree. This would mean that it's another state 
- (refactor) I'd rather not traverse the full tree map all time when toggling items, it's not really needed. 
- (refactor) The list service becomes a bit busy, i'd move tree related stuff to tree service.

Reply:
I'd also be in favour of changing the weird behaviour of:
- not having a toggle icon on the root item
- not collapsing all tree nodes, when using collapse all; it's a confusing experience.

Reply:
And:
- (ux) collapsing all tree nodes, including root children
- (configurability) make the (initial) `minimalExpanded` children extensible.

Reply:
I've done an optimisation at #9007

Reply:
> (issue) The active tree item is not expanded if you navigate browser history

thats a good point, didnt verify that before, sorry

> (issue) The active item is also expanded, but it shouldn't imho. we should only open the tree until the active item instead.

yeah, that was unexpected. Issue probably created while last refactor

> not having a toggle icon on the root item
> not collapsing all tree nodes, when using collapse all; it's a confusing experience.
> (ux) collapsing all tree nodes, including root children

Yeah, I don't like it too but it was done in accordance with business requirements

> (configurability) make the (initial) minimalExpanded children extensible.

I think it was done by `initialExpanded = 1;` property